packages/store/addon/-private/system/store/finders.js

```diff
@@ -262,9 +262,19 @@ function fixRelationshipData(relationshipData, relationshipKind, { id, modelName
   if (relationshipKind === 'hasMany') {
     payload = relationshipData || [];
     if (relationshipData) {
-      if (
-        !relationshipData.filter((v) => v.type === parentRelationshipData.type && v.id === parentRelationshipData.id)[0]
-      ) {
+      // IE11 does not support array.find
+      // these arrays could be massive so this is better than filter
+      // Note: this is potentially problematic if type/id are not in the
+      // same state of normalization.
+      let found = false;
+      for (let i = 0; i < relationshipData.length; i++) {
```

Comment:
Not sure what your definition of massive is, but once you get in to the 10k+ range this optimization can become worthwhile
```suggestion
      for (let i = 0, len = relationshipData.length; i < len; i++) {
```

Reply:
I'm under the impression though @krisselden can correct me that most browsers now optimize length access in this way anyway.